source/slang/core.meta.slang

```diff
@@ -1419,6 +1451,19 @@ for( int C = 2; C <= 4; ++C )
         if(rr == R && cc == C) continue;
         sb << "__intrinsic_op(" << int(kIROp_MatrixReshape) << ") __init(matrix<T," << rr << "," << cc << ", L> value);\n";
     }
+
+    // Initialize from matrix and vector
+    if (R > 2)
```

Comment:
If R==4, we shouldn't define anything here, right?

Reply:
When R is 4, we will have to have the followings:
```
__generic<T, let L:int> __extension matrix<T, 4,2, L>
{
__init(matrix<T,3,2> m, vector<T,2> row3) { this = This(m[0], m[1], m[2], row3); }
}
__generic<T, let L:int> __extension matrix<T, 4,3, L>
{
__init(matrix<T,3,3> m, vector<T,3> row3) { this = This(m[0], m[1], m[2], row3); }
}
__generic<T, let L:int> __extension matrix<T, 4,4, L>
{
__init(matrix<T,3,4> m, vector<T,4> row3) { this = This(m[0], m[1], m[2], row3); }
}
```
It is little confusing but what we shouldn't have is when R is 1 where it should just take a vector.

Reply:
Oh, I see what's going on here, my bad. This is good then. In general HLSL might be allowing constructing a matrix with arbitrary vectors + matrix parameters and just concatenatibf them together. But we don't have to add those in this PR if the epic shader is not using them.